File: backend/app/schemas/legal.py
"""Public legal document API schemas (issue #321)."""

from __future__ import annotations

from pydantic import BaseModel, Field


class LegalDocumentSummary(BaseModel):
    id: str
    title: str
    version: str
    updated_at: str = Field(alias="updatedAt")
    languages: list[str]

    model_config = {"populate_by_name": True}


class LegalCatalogResponse(BaseModel):
    version: str
    documents: list[LegalDocumentSummary]

    model_config = {"populate_by_name": True}


class LegalDocumentResponse(BaseModel):
    id: str
    title: str
    version: str
    updated_at: str = Field(alias="updatedAt")
    lang: str
    markdown: str

    model_config = {"populate_by_name": True}

File: mobile/src/types/legal.ts
export type LegalDocumentId = 'terms' | 'privacy' | 'acceptable-use';

export type LegalDocumentSummary = {
  id: LegalDocumentId | string;
  title: string;
  version: string;
  updatedAt: string;
  languages: string[];
};

export type LegalCatalog = {
  version: string;
  documents: LegalDocumentSummary[];
};

export type LegalDocument = {
  id: LegalDocumentId | string;
  title: string;
  version: string;
  updatedAt: string;
  lang: string;
  markdown: string;
};
